### DISCUSSION TITLE
FetchError: request to https://.../api/v1/projects failed, reason: connect ETIMEDOUT 34.241.115.67:443

### DISCUSSION THREAD

**Discussion by AIEngKenn** [vercel/next.js#31973] [Help]
### What version of Next.js are you using?

^12.0.4

### What version of Node.js are you using?

16.13.0

### What browser are you using?

Google Chrome

### What operating system are you using?

Windows

### How are you deploying your application?

next export

### Describe the Bug

Getting this error,
I don't really understand why. I really want your help 🙏

My API link: https://mvp4startup-api.herokuapp.com/api/v1/projects

```
export async function getStaticProps() {
  const res = await fetch(`${server}/api/v1/projects`);
  const projects = await res.json();
  const formations = await fetch(`${server}/api/v1/formations`).then((res) =>
    res.json()
  );
  const businessplans = await fetch(`${server}/api/businessplans`).then((res) =>
    res.json()
  );
  return {
    props: {
      projects,
      formations,
      businessplans,
    },
  };
}
```

### Expected Behavior

I expect to receive data and render it to the page, through props.

### To Reproduce

Setup next app (version ^12.0.4) and make a getStaticProps call.

**Comment by MonstraG** [comment]
Same code but actually readable; use triple quotes and language name for it to work: \```ts 
```ts
export async function getStaticProps() {
	const res = await fetch(`${server}/api/v1/projects`);
	const projects = await res.json();
	const formations = await fetch(`${server}/api/v1/formations`).then((res) => res.json());
	const businessplans = await fetch(`${server}/api/businessplans`).then((res) => res.json());
	return {
		props: { projects, formations, businessplans }
	};
}
```

I'm guessing that `${server}` resolves to IP you've posted in the title, are you sure your server is running?

**Comment by AIEngKenn** [comment]
Actually, it's a Remote Server running at [https://mvp4startup-api.herokuapp.com/api/v1/projects](https://mvp4startup-api.herokuapp.com/api/v1/projects), [https://mvp4startup-api.herokuapp.com/api/v1/formations](https://mvp4startup-api.herokuapp.com/api/v1/formations) & [https://mvp4startup-api.herokuapp.com/api/v1/businessplans](https://mvp4startup-api.herokuapp.com/api/v1/businessplans)


In Local, I've created a Config Folder (Config/index.js) where I've typed this:

```
const dev = process.env.NODE_ENV !== "production";

export const server = dev
  ? "https://mvp4startup-api.herokuapp.com"
  : "https://mvp4startup-api.herokuapp.com";
```

And the server is called from the file. It was working fine, just mast week, when I got back to the project, I got this error :(

**Comment by MonstraG** [comment]
this:
```ts
export async function getStaticProps() {
	const server = "https://mvp4startup-api.herokuapp.com";
	const res = await fetch(`${server}/api/v1/projects`);
	const projects = await res.json();
	const formations = await fetch(`${server}/api/v1/formations`).then((res) => res.json());
	return {
		props: {
			projects,
			formations
		}
	};
}
```
works fr me and I can succesfully receive both projects and formations.

Including businessplans gives me `FetchError: invalid json response body at https://mvp4startup-api.herokuapp.com/api/businessplans reason: Unexpected end of JSON input`

In any case, it doesn't seem like its an issue with nextjs.

**Comment by AIEngKenn** [comment]
Sorry, check with this new config: 

```
export async function getStaticProps() {
	const server = "https://mvp4startup-api.herokuapp.com/api/v1";
	const res = await fetch(`${server}/projects`);
	const projects = await res.json();
	const formations = await fetch(`${server}/formations`).then((res) => res.json());
	return {
		props: {
			projects,
			formations
		}
	};
}
```

```
const dev = process.env.NODE_ENV !== "production";

export const server = dev
  ? "https://mvp4startup-api.herokuapp.com/api/v1"
  : "https://mvp4startup-api.herokuapp.com/api/v1";
```

just added the `/api/v1` in the config file

**Comment by MonstraG** [comment]
Try making curl request to the api yourself to make sure that you can reach it.

**Comment by AIEngKenn** [comment]
just checked using postman [here](https://web.postman.co/workspace/My-Workspace~9e9ddbf7-41a6-416f-85e4-684f1c37ddf9/request/create?requestId=c3959c3a-92bd-4e8d-87d8-25f56ad484e7)

It works fine, I've got the JSON data

But I still got this error while fetching data :(

**Comment by jeanbmar** [comment]
@KennStack01 I had the same issue, fetch was hanging forever and it was related to some canvas generation issue when doing SSR. Vercel fetch function didn't throw for some reason

**Comment by bsor-dev** [comment]
Same issue. Working properly in local but got same issue in production using fetch api in middleware. I'm using Next v12.1.6, proxy pass nginx. But my other app has the same config but the fetch is working properly in production

**Reply by svbackend** [reply]
Did you fix it? I have the same problem, works fine locally, api endpoint works 100% because i'm calling it from anroid app and it works, while nextjs getServerSideProps() just gives an error "ETIMEDOUT"

**Reply by javierfuentesm** [reply]
> Did you fix it? I have the same problem, works fine locally, api endpoint works 100% because i'm calling it from android app and it works, while nextjs getServerSideProps() just gives an error "ETIMEDOUT"

Did any of you fix it? im getting the same error, I can reach the endpoint using Postman or the browser it only fails when I call the endpoint from my bot of discord

**Reply by javierfuentesm** [reply]
@bsor-dev @svbackend
